src/wrappers/OsipiBase.py

```diff
@@ -83,7 +83,13 @@ def osipi_fit(self, data, bvalues, **kwargs):
         
         #args = [data, use_bvalues, use_initial_guess, use_bounds, use_thresholds]
         #args = [arg for arg in args if arg is not None]
-        results = self.ivim_fit(*args, **kwargs)
+
+        # Assuming the last dimension of the data is the signal values of each b-value
+        results = np.empty(list(data.shape[:-1])+[3]) # Create an array with the voxel dimensions + the ones required for the fit
+        for ijk in tqdm(np.ndindex(data.shape[:-1]), total=np.prod(data.shape[:-1])):
```

Comment:
I'm not sure if we have algorithms that accept 2+D inputs, but at some point we'll need to iterate on the correct number of dimensions here. E.g. if an algorithm is 2D, this loop should only operate on the final dimension for that algorithm.